### DISCUSSION TITLE
1.5.0-beta3

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#343] [Обновления]
⚠ Это бета версия и она может работать нестабильно. Обо всех проблемах рекомендуется сразу же сообщать в [Issues](https://github.com/ilyhalight/voice-over-translation/issues)

## Список изменений
- Добавлена экспериментальная поддержка трансляций (Необходимо расширение для отключение CORS. С задержкой как повезет, но в большинстве случаев довольно терпимо +-0-2 секунды от оригинала, но иногда бывает отставание до 10 секунд, перезапуск перевода помогает немного уменьшить задержку)
- Добавлен HLS.js
- Теперь, в режиме разработки расширение имеет префикс "[DEBUG]"
- Исправлено получение блоба заместо ссылки на видео в Udemy

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.5.0-beta2...1.5.0-beta3

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.5.0-beta3'>1.5.0-beta3</a>.</em>

**Comment by ilyhalight** [comment]
Пока что, наверное, будет так, все же, это лучше чем ничего. Но это, все равно, далеко от оригинального перевода трансляций в ябраузере

**Reply by NullVerdict** [reply]
Стоп, зачем отключать CORS в Cloudflare версии, если можно передавать содержимое напрямую, без посредника.

Также HLS.JS можно попробывать заменить на hls-web-audio.

**Reply by ilyhalight** [reply]
cors в обоих версиях надо отключать, иначе аудио трансляция даже не будет загружена, сам их сервер не поддерживает cors запросы
![image](https://github.com/ilyhalight/voice-over-translation/assets/62353659/c11731a0-ac27-4cc6-99b8-db786916ce0d)

> если можно передавать содержимое напрямую, без посредника.

немного не понял про что ты имеешь в виду

**Reply by NullVerdict** [reply]
> cors в обоих версиях надо отключать, иначе аудио трансляция даже не будет загружена, сам их сервер не поддерживает cors запросы ![image](https://user-images.githubusercontent.com/62353659/270451169-c11731a0-ac27-4cc6-99b8-db786916ce0d.png)
> 
> > если можно передавать содержимое напрямую, без посредника.
> 
> немного не понял про что ты имеешь в виду

Я имел в виду полное проксирование m3u8-потоков, от плейлиста до аудиодорожек в нем. Но я думаю это повысит задержку на 1-3 секунды. (может быть меньше)

**Reply by ilyhalight** [reply]
Я пробовал проксировать через [m3u8-proxy](https://github.com/Eltik/M3U8-Proxy), но это не помогло. Возникает, таже, ошибка.
selfhosted: 
![image](https://github.com/ilyhalight/voice-over-translation/assets/62353659/41d08c56-775e-499d-8086-2d578aa954f8)

public (похоже забыли поменять домен и из-за этого не работает):
![image](https://github.com/ilyhalight/voice-over-translation/assets/62353659/7f980548-6340-44d5-8f0e-b1af9c3d2707)

**Reply by NullVerdict** [reply]
Попробуй [это](https://github.com/Gratenes/m3u8CloudflareWorkerProxy)

**Reply by ilyhalight** [reply]
> Попробуй [это](https://github.com/Gratenes/m3u8CloudflareWorkerProxy)

с этим ошибки cors нету, но почему-то с ним багается ссылка при запросе через hls (именно при запросе к яндексу, с другими сайтами все нормально работает)
![image](https://github.com/ilyhalight/voice-over-translation/assets/62353659/0aeb3837-2784-409c-b4ff-c29e5aafb39c)

**Reply by NullVerdict** [reply]
> > Попробуй [это](https://github.com/Gratenes/m3u8CloudflareWorkerProxy)
> 
> с этим ошибки cors нету, но почему-то с ним багается ссылка при запросе через hls (именно при запросе к яндексу, с другими сайтами все нормально работает) ![image](https://user-images.githubusercontent.com/62353659/270660290-0aeb3837-2784-409c-b4ff-c29e5aafb39c.png)

А у Яндекса есть привязка по айпи к плейлисту?

**Reply by ilyhalight** [reply]
> > > Попробуй [это](https://github.com/Gratenes/m3u8CloudflareWorkerProxy)
> > 
> > 
> > с этим ошибки cors нету, но почему-то с ним багается ссылка при запросе через hls (именно при запросе к яндексу, с другими сайтами все нормально работает) ![image](https://user-images.githubusercontent.com/62353659/270660290-0aeb3837-2784-409c-b4ff-c29e5aafb39c.png)
> 
> А у Яндекса есть привязка по айпи к плейлисту?

нет.

Разобрался в чем проблема, оказывается в воркере всегда использовался юрл из первого запроса из-за чего ссылка на плейлист багалась. Чуть-чуть модифицировав воркер все заработало

**Reply by NullVerdict** [reply]
> > > > Попробуй [это](https://github.com/Gratenes/m3u8CloudflareWorkerProxy)
> > > 
> > > 
> > > с этим ошибки cors нету, но почему-то с ним багается ссылка при запросе через hls (именно при запросе к яндексу, с другими сайтами все нормально работает) ![image](https://user-images.githubusercontent.com/62353659/270660290-0aeb3837-2784-409c-b4ff-c29e5aafb39c.png)
> > 
> > 
> > А у Яндекса есть привязка по айпи к плейлисту?
> 
> нет.
> 
> Разобрался в чем проблема, оказывается в воркере всегда использовался юрл из первого запроса из-за чего ссылка на плейлист багалась. Чуть-чуть модифицировав воркер все заработало

Прикольный баг) А что по задержке? +- одинаково?

**Reply by ilyhalight** [reply]
> Прикольный баг) А что по задержке? +- одинаково?

Задержка, примерно, такая же. Но иногда бывает что аудио может чуть-чуть залагать, бувально на пол секунды

**Reply by NullVerdict** [reply]
> > Прикольный баг) А что по задержке? +- одинаково?
> 
> Задержка, примерно, такая же. Но иногда бывает что аудио может чуть-чуть залагать, бувально на пол секунды

А что по поводу HLS.JS и hls-web-audio? В скрипте кстати используется старый HLS.JS.

**Reply by ilyhalight** [reply]
> А что по поводу HLS.JS и hls-web-audio? 

https://github.com/sound-ws/hls-web-audio ?

**Reply by NullVerdict** [reply]
> > А что по поводу HLS.JS и hls-web-audio?
> 
> https://github.com/sound-ws/hls-web-audio ?

Да, я пока другой альтернативы не нашел.

**Reply by ilyhalight** [reply]
> > > А что по поводу HLS.JS и hls-web-audio?
> > 
> > 
> > https://github.com/sound-ws/hls-web-audio ?
> 
> Да, я пока другой альтернативы не нашел.

попробую

**Reply by NullVerdict** [reply]
Кстати, кнопка В ЭФИРЕ не синхронизирует стрим с аудио. Ну или как в яндексе сделать две кнопки.

**Reply by NullVerdict** [reply]
> > > > А что по поводу HLS.JS и hls-web-audio?
> > > 
> > > 
> > > https://github.com/sound-ws/hls-web-audio ?
> > 
> > 
> > Да, я пока другой альтернативы не нашел.
> 
> попробую

Cloudflare добавил Whisper в воркеры. Не знаю только какой максимальная длительность поддерживается.

**Comment by FiLinX** [comment]
> * Необходимо расширение для отключение CORS

а не подскажете - где взять то его? )

**Reply by NullVerdict** [reply]
> > * Необходимо расширение для отключение CORS
> 
> а не подскажете - где взять то его? )

CORS Everywhere в FireFox, для хромоподобных [CORS Unblock](https://chrome.google.com/webstore/detail/cors-unblock/lfhmikememgdcahcdlaciloancbhjino)

**Comment by Michael1297** [comment]
Перевод трансляций работает? Я попробовал включить его на нескольких стримах на ютубе: перевод активируется и при этом звука нет.

**Reply by NullVerdict** [reply]
> Перевод трансляций работает? Я попробовал включить его на нескольких стримах на ютубе: перевод активируется и при этом звука нет.

CORS отключи

**Reply by Michael1297** [reply]
Сейчас работает. Возможно что сервера яндекса подлагивали в тот момент либо мой интернет 💩.

**Reply by Michael1297** [reply]
Кстати оно что не может переводить стримы с японского?

**Reply by NullVerdict** [reply]
> Кстати оно что не может переводить стримы с японского?

Проверь

**Reply by Michael1297** [reply]
я попробовал несколько стримов и у всех ошибка "There is no connection to the server"
